examples/training/semantic_segmentation/pascal_voc/fcn.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Title: Train an Semantic Segmentation Model on Pascal VOC 2012 using KerasCV
+Author: [tanzhenyu](https://github.com/tanzhenyu)
+Date created: 2022/10/25
+Last modified: 2022/10/25
+Description: Use KerasCV to train a FCN on Pascal VOC 2012.
+"""
+
+import sys
+
+import tensorflow as tf
+from absl import flags
+from absl import logging
+
+from keras_cv.datasets.pascal_voc.segmentation import load
+from keras_cv.models.segmentation.fcn import FCN
+
+flags.DEFINE_string(
+    "weights_path",
+    "weights_{epoch:02d}.h5",
+    "Directory which will be used to store weight checkpoints.",
+)
+flags.DEFINE_boolean(
+    "mixed_precision",
+    True,
+    "Whether or not to use FP16 mixed precision for training.",
+)
+flags.DEFINE_string(
+    "tensorboard_path",
+    "logs",
+    "Directory which will be used to store tensorboard logs.",
+)
+FLAGS = flags.FLAGS
+FLAGS(sys.argv)
+
+if FLAGS.mixed_precision:
+    logging.info("mixed precision training enabled")
+    tf.keras.mixed_precision.set_global_policy("mixed_float16")
+
+# Try to detect an available TPU. If none is present, default to MirroredStrategy
+try:
+    tpu = tf.distribute.cluster_resolver.TPUClusterResolver.connect()
+    strategy = tf.distribute.TPUStrategy(tpu)
+except ValueError:
+    # MirroredStrategy is best for a single machine with one or multiple GPUs
+    strategy = tf.distribute.MirroredStrategy()
+print("Number of accelerators: ", strategy.num_replicas_in_sync)
+
+# parameters from FasterRCNN [paper](https://arxiv.org/pdf/1506.01497.pdf)
```

Comment:
Should we use the same params as from the FasterRCNN paper here? What do the FCN paper params look like?

Reply:
No, the parameters differ from FasterRCNN, LR and batch size in the case of fcn, which I will change.